spec/category_spec.rb

```diff
@@ -828,6 +828,23 @@
         expect(@dv[:yoda]).to eq(666)
       end
 
+      it "assigns a new category at a new index " do
+        @dv.add_category :abc
+        @dv.send(:set, :xtbr, :abc)
+        expect(@dv[:xtbr]).to eq(:abc)
+      end
+
+      it "assigns a new category at a multiple indices " do
+        @dv.add_category :rbn
+          @dv.send(:set, [:padme, :bdn], :rbn)
+          @dv.send(:set, :cy..:db, :rbn)
+        expect(@dv[:bdn]).to eq(:rbn)
```

Comment:
It's good idea to test ranges and arrays in different tests. In general always test two different things in different tests.

Reply:
Sure, please check.

Reply:
ping! @lokeshh